Added default values & color/B&W print option

pums12 opened the pull request:
Set default values for PrintNodePrintJob and PrintNodePrintJobOptions

Added Print method to PrintNodeJob, so that you can print without having the query for the printers first if you already have the PrinterId.

Added PrintNodePrintJob constructor

christianz commented on PrintNodePrintJobOptions.cs:
```diff
@@ -73,7 +73,7 @@ public class PrintNodePrintJobOptions
         /// </summary>
         [JsonProperty("pages")]
-        public string Pages { get; set; }
+        public string Pages { get; set; } = "-";
```
Have you tested this? I haven't looked very well, but can't find anything about setting Pages to "-" in the official documentation?

pums12 replied:
Yes, I have tested it but only with a single page.

It is documented in Print Job options under Pages  "- prints all pages."
https://www.printnode.com/docs/api/curl#printjob-creating

christianz commented on PrintNodePrintJobOptions.cs:
```diff
@@ -64,7 +64,7 @@ public class PrintNodePrintJobOptions
         /// </summary>
         [JsonProperty("nup")]
-        public int Nup { get; set; }
+        public int Nup { get; set; } = 1;
```
According to the official docs this should be an array of integers, so we might have another issue here. What's the reasoning/result of defaulting this to 1?

pums12 replied:
The print capabilities return an array for Nup but for the print, it asks
for an Integer. The default they document is 1.

christianz commented on PrintNodePrintJob.cs:
```diff
@@ -136,6 +147,18 @@ public sealed class PrintNodePrintJob
             return list.FirstOrDefault();
+        }
+
+		 public async Task<long> Print()
```
Is this necessary when you can use PrintNodePrinter.AddPrintJob()?

pums12 replied:
I wanted to be able to print a job without having to query PrintNode for
the printer first.

I already had the ID from the Print Dialogue I had shown the user.

christianz commented:
Hi, and thank you so much for the PR!

I don't have much time left over to maintain this repo these days, but it's awesome to see that people are using it and are still adding to it. I've added some comments asking about your reasoning on some of the changes.

pums12 commented:
Yes, I have tested it but only with a single page.

It is documented in Print Job options under Pages  "- prints all pages."
https://www.printnode.com/docs/api/curl#printjob-creating

On Thu, 7 Jun 2018 at 12:56, Christian Zachariasen <notifications@github.com>
wrote:

> *@christianz* commented on this pull request.
> ------------------------------
>
> In PrintNodePrintJobOptions.cs
> <https://github.com/christianz/printnode-net/pull/5#discussion_r193719729>
> :
>
> > @@ -73,7 +73,7 @@ public class PrintNodePrintJobOptions
>          /// See <a href="https://www.printnode.com/docs/api/curl/#parameters">Parameters</a>.
>          /// </summary>
>          [JsonProperty("pages")]
> -        public string Pages { get; set; }
> +        public string Pages { get; set; } = "-";
>
> Have you tested this? I haven't looked very well, but can't find anything
> about setting Pages to "-" in the official documentation?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/christianz/printnode-net/pull/5#pullrequestreview-126746679>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGQE1bHVn1G_aDJ5pU6PSk3CqLvdI5Rdks5t6RTggaJpZM4UZtjz>
> .
>

pums12 commented:
The print capabilities return an array for Nup but for the print, it asks
for an Integer. The default they document is 1.

On Thu, 7 Jun 2018 at 12:57, Christian Zachariasen <notifications@github.com>
wrote:

> *@christianz* commented on this pull request.
> ------------------------------
>
> In PrintNodePrintJobOptions.cs
> <https://github.com/christianz/printnode-net/pull/5#discussion_r193720114>
> :
>
> > @@ -64,7 +64,7 @@ public class PrintNodePrintJobOptions
>          /// See <a href="https://www.printnode.com/docs/api/curl/#printer-capabilities">Printer capabilities</a>.
>          /// </summary>
>          [JsonProperty("nup")]
> -        public int Nup { get; set; }
> +        public int Nup { get; set; } = 1;
>
> According to the official docs this should be an array of integers, so we
> might have another issue here. What's the reasoning/result of defaulting
> this to 1?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/christianz/printnode-net/pull/5#pullrequestreview-126747152>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGQE1UA4MJziUCmiyseTrd9MnyU0A7fJks5t6RVCgaJpZM4UZtjz>
> .
>

pums12 commented:
I wanted to be able to print a job without having to query PrintNode for
the printer first.

I already had the ID from Print Dialogue I had shown the user.

On Thu, 7 Jun 2018 at 13:00, Christian Zachariasen <notifications@github.com>
wrote:

> *@christianz* commented on this pull request.
> ------------------------------
>
> In PrintNodePrintJob.cs
> <https://github.com/christianz/printnode-net/pull/5#discussion_r193720860>
> :
>
> > @@ -136,6 +147,18 @@ public sealed class PrintNodePrintJob
>              var list = JsonConvert.DeserializeObject<IEnumerable<IEnumerable<PrintNodePrintJobState>>>(response);
>
>              return list.FirstOrDefault();
> +        }
> +
> +		 public async Task<long> Print()
>
> Is this necessary when you can use PrintNodePrinter.AddPrintJob()?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/christianz/printnode-net/pull/5#pullrequestreview-126748040>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGQE1drk2GWKF5_57R52vk0VfH8LG1-3ks5t6RX4gaJpZM4UZtjz>
> .
>

pums12 commented:
Hi,

No problem.

Thanks for taking the time to build the library in the first place!

BTW: The Color option I have added for PrintJobs is undocumented on
PrintNode but they mentioned I could try in a support email and so far it
seems to work well (only on Print Engine 6 though).

Thanks,

Ian

On Thu, 7 Jun 2018 at 13:02, Christian Zachariasen <notifications@github.com>
wrote:

> Hi, and thank you so much for the PR!
>
> I don't have much time left over to maintain this repo these days, but
> it's awesome to see that people are using it and are still adding to it.
> I've added some comments asking about your reasoning on some of the lines.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/christianz/printnode-net/pull/5#issuecomment-395395805>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGQE1Z5F1_X4U_tlHPOARm_yQlhyC7ghks5t6RZSgaJpZM4UZtjz>
> .
>
